python/lsst/ts/ess/csc/rpi_data_client.py

```diff
@@ -34,16 +34,19 @@
 from lsst.ts import salobj, tcpip
 from lsst.ts.ess import common
 
-# Time limit for connecting to the RPi (seconds)
+# Time limit for connecting to the RPi (seconds).
 CONNECT_TIMEOUT = 5
 
-# Time limit for communicating with the RPi (seconds)
-# This includes writing a command and reading the response
-# and reading telemetry (seconds)
-COMMUNICATE_TIMEOUT = 5
+# The number of timeouts to allow before raising a TimeoutError.
+NUM_ALLOWED_TIMEOUTS = 5
 
 
 class RPiDataClient(common.BaseDataClient):
+    # Timeout limit for communicating with the RPi (seconds). This includes
+    # writing a command and reading the response and reading telemetry. Unit
+    # tests can set this to a lower value to speed up the test.
+    communicate_timeout = 60
```

Comment:
It seems a bit odd to make communicate_timeout a class variable and num_allowed_timeouts a module constant, when they work hand-in-hand.

Note that unit tests can override module constants, though I think it requires a mock to do it politely. (Mocks are a nice way to do this in any case, since they don't affect other code if tests are run in parallel).

In any case, a small comment that may not be worth doing anything about. What you did works and we don't yet run tests in parallel (though I hope to change that if we switch to Kafka).

Reply:
Good point and thanks for pointing out mocking. I made it a module constant to be more consistent with the other constants.